ibis/backends/tests/test_join.py

```diff
@@ -181,7 +181,10 @@ def test_mutate_then_join_no_column_overlap(batting, awards_players):
         param(lambda left, right: left.join(right, "year", how="semi"), id="how_semi"),
     ],
 )
-def test_semi_join_topk(batting, awards_players, func):
+def test_semi_join_topk(con, batting, awards_players, func):
+    if con.name == "sqlite":
+        # TODO: remove after CTE extraction is reimplemented
```

Comment:
CTE extraction has been merged, so I think this can be removed.

Reply:
I just checked out the PR, and this is still very slow so perhaps CTE extraction isn't sufficient to address the issue or we're not extracting enough?